demos/python_demos/image_inpainting_demo/inpainting.py

```diff
@@ -0,0 +1,102 @@
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import os
+import cv2
+import numpy as np
+
+from openvino.inference_engine import IENetwork, IECore
+
+class ImageInpainting(object):
+    def __init__(self, ie, model_path, device='CPU', parts=8, max_brush_width=24, max_length=100, max_vertex=20):
+        model = IENetwork(model=model_path, weights=os.path.splitext(model_path)[0] + '.bin')
+
+        assert len(model.inputs) == 2, "Expected 2 input blob"
+        assert len(model.outputs) == 1, "Expected 1 output blobs"
+
+        self._input_layer_names = sorted(model.inputs)
+        self._output_layer_name = next(iter(model.outputs))
+
+        self._ie = ie
+        self._exec_model = self._ie.load_network(model, device)
+        self.infer_time = -1
+
+        _, channels, input_height, input_width = model.inputs[self._input_layer_names[0]].shape
+        assert channels == 3, "Expected 3-channel input"
+
+        _, channels, mask_height, mask_width = model.inputs[self._input_layer_names[1]].shape
+        assert channels == 1, "Expected 1-channel input"
+
+        assert mask_height == input_height and mask_width == input_width, "Mask size expected to be equal to image size"
+        self.input_height = input_height
+        self.input_width = input_width
+
+        self.parts = parts
+        self.max_brush_width = max_brush_width
+        self.max_length = max_length
+        self.max_vertex = max_vertex
+
+    @staticmethod
+    def _free_form_mask(max_vertex, max_length, max_brush_width, h, w, max_angle=360):
+        mask = np.zeros((h, w, 1), np.float32)
+        num_vertex = np.random.randint(max_vertex + 1)
+        start_y = np.random.randint(h)
+        start_x = np.random.randint(w)
+        brush_width = 0
+        for i in range(num_vertex):
+            angle = np.random.randint(max_angle + 1)
+            angle = angle / 360.0 * 2 * np.pi
+            if i % 2 == 0:
+                angle = 2 * np.pi - angle
+            length = np.random.randint(max_length + 1)
+            brush_width = np.random.randint(10, max_brush_width + 1) // 2 * 2
+            next_y = start_y + length * np.cos(angle)
+            next_x = start_x + length * np.sin(angle)
+
+            next_y = np.maximum(np.minimum(next_y, h - 1), 0).astype(np.int)
+            next_x = np.maximum(np.minimum(next_x, w - 1), 0).astype(np.int)
+            cv2.line(mask, (start_y, start_x), (next_y, next_x), 1, brush_width)
+            cv2.circle(mask, (start_y, start_x), brush_width // 2, 2)
+
+            start_y, start_x = next_y, next_x
+        cv2.circle(mask, (start_y, start_x), brush_width // 2, 2)
```

Comment:
Given that this draws the start vertex, it would make more sense if it was done before the loop.